app/src/main/java/com/example/wellfed/mealplan/MealBookFragment.java

```diff
@@ -1,18 +1,68 @@
+/*
+ * MealBookFragment
+ *
+ * Version: v1.0.0
+ *
+ * Date: 2022-10-24
+ *
+ * Copyright notice:
+ * This file is part of well-fed.
+ *
+ * well-fed is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * well-fed is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with well-fed. If not, see <https://www.gnu.org/licenses/>.
+ */
 package com.example.wellfed.mealplan;
 
+import android.graphics.Typeface;
 import android.os.Bundle;
+import android.text.SpannableStringBuilder;
+import android.text.style.StyleSpan;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.TextView;
 
+import androidx.activity.result.ActivityResultCallback;
+import androidx.activity.result.ActivityResultLauncher;
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
 
 import com.example.wellfed.R;
 
-public class MealBookFragment extends Fragment {
+public class MealBookFragment extends Fragment implements MealPlanAdapter.Launcher {
+    private TextView userFirstNameTextView;
+    private TextView callToActionTextView;
+    private RecyclerView mealPlanRecyclerView;
+    private MealPlanAdapter mealPlanAdapter;
+    private MealPlanController controller;
+    private int position;
+
+    ActivityResultLauncher<MealPlan> launcher = registerForActivityResult(new
+                    MealPlanContract(),
+            new ActivityResultCallback<MealPlan>() {
+                @Override
+                public void onActivityResult(MealPlan result) {
+                    if (result == null) {
```

Comment:
can you actually look into it or mark it as an issue, since as of right now our app deletes an item from the list if a null is retuned which is totally possible to be returned as null when ActivityResult is not okay. can you think of a better design?

Reply:
I'll look into it later